common4j/src/main/com/microsoft/identity/common/java/flighting/CommonFlightsManager.kt

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.flighting
+
+import com.microsoft.identity.common.java.logging.Logger
+import org.json.JSONObject
+
+/**
+ * Consumer of commons needs to implement [IFlightsManager] interface
+ * and set it using CommonFlightManager.initializeCommonFlightsManager(@NonNull IFlightsManager flightsManager)
+ * to provide Flight Values for CommonFlights
+ * If no Flight Provider is set, default value of the flight will be used
+ */
+object CommonFlightsManager : IFlightsManager {
+    private val TAG = CommonFlightsManager::class.java.simpleName
+    private var mFlightsManager: IFlightsManager = DefaultValueFlightsManager
+
+    fun initializeCommonFlightsManager(flightsManager: IFlightsManager) {
```

Comment:
Does this mean we can use flights in common and MSAL now?

Reply:
Broker and MSAL can inject this to have common code their respective flights. Since MSAL does not have any flights, default would be used. Broker injects its own flights manager, so flight in common code have broker flight.